src/lib/ts/cache-compiler-host.ts

```diff
@@ -289,3 +288,15 @@ export function augmentProgramWithVersioning(program: ts.Program): void {
     return files;
   };
 }
+
+export function invalidateProgramFiles(program: ts.Program, files: Set<string>): void {
+  files.forEach(file => {
+    const source = program.getSourceFile(file) as ts.SourceFile & { version?: string };
+
+    if (source) {
+      source.version = createHash('sha256')
+        .update(source.text + new Date().getTime())
```

Comment:
I don't quite understand the intent here. Adding the current time as part of the digest would mean the source file's content is mostly irrelevant. In any case, it would be better to split the string concatenation into two independent `update` calls.

I'd have to check how the `ts.SourceFile.version` is treated internally for better insight into what this means exactly.